tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/WriteSourceRepoProperties.cs

```diff
@@ -151,20 +150,36 @@ private static DerivedVersion GetVersionInfo(string version, string commitCount)
             throw new FormatException($"Can't derive a build ID from version {version} (commit count {commitCount})");
         }
 
-        private static string GetRepoNameOrDefault(Dependency dependency)
+        private static string GetDefaultRepoNameFromUrl(string repoUrl)
         {
-            if (dependency.RepoName != null)
-            {
-                return dependency.RepoName;
-            }
-            var repoUrl = dependency.Uri;
             if (repoUrl.EndsWith(".git"))
             {
                 repoUrl = repoUrl.Substring(0, repoUrl.Length - ".git".Length);
             }
             return repoUrl.Substring(repoUrl.LastIndexOf("/") + 1);
         }
 
+        private static string GetRepoNameOrDefault(Dependency dependency)
+        {
+            if (dependency.RepoName != null)
+            {
+                return dependency.RepoName;
+            }
+            return GetDefaultRepoNameFromUrl(dependency.Uri);
```

Comment:
Thanks, looking clear.

Noticed this makes this particular method simple enough for:

```C#
return dependency.RepoName ?? GetDefaultRepoNameFromUrl(dependency.Uri);
```

Maybe not worth resetting PR validation for though. 😄

Reply:
If something breaks in the PR I'll add it with the fix though.  Thanks!